packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js

```diff
@@ -192,7 +192,7 @@ const reducer = (state, action) =>
           set(draftState, path, [value]);
         } else {
           const modifiedDataRelations = get(state, path);
-          const newRelations = [...modifiedDataRelations, value];
+          const newRelations = modifiedDataRelations ? [...modifiedDataRelations, value] : [value];
```

Comment:
I don't think `modifiedDataRelations` should ever not be an `array` when it's not a `toOneRelation`, so while this fixes the issue, I don't think it's the right place for the fix...

Also you've not added tests for this change, but like I said, we should be looking at _why_ `modifiedDataRelations` is not an `array`.

Reply:
Is the fix such that `toOneRelation` is `true`?

Reply:
If you're using an XtoOne relation then it shouldn't be doing this. To have a fix we need to know _why_ `modifiedDataRelations` is `undefined`

Reply:
Thank you so much. I understand the cause of the `undefined`. The `toOneRelation` condition is not applicable, please check this fix.